dbms/src/DataStreams/ParallelAggregatingBlockInputStream.cpp

```diff
@@ -273,12 +269,12 @@ void ParallelAggregatingBlockInputStream::execute()
     if (total_src_rows == 0 && params.keys_size == 0 && !params.empty_result_for_aggregation_by_empty_set)
     {
         auto & data = *many_data[0];
-        aggregator.executeOnBlock(
-            children.at(0)->getHeader(),
-            data,
-            threads_data[0].key_columns,
-            threads_data[0].aggregate_columns,
-            0);
+        auto & agg_process_info = threads_data[0].agg_process_info;
+        RUNTIME_CHECK_MSG(
+            agg_process_info.start_row == agg_process_info.end_row,
+            "Previous block is not processed yet");
+        agg_process_info.resetBlock(children.at(0)->getHeader());
```

Comment:
How about move CHECK into resetBlock? Maybe some prepare logic can be moved also?

Reply:
Done